bot/utils/database-helper.js

```diff
@@ -51,4 +44,45 @@ module.exports.openDatabase = () => {
  */
 module.exports.getDatabase = () => {
     return database;
-};
\ No newline at end of file
+};
+
+/**
+ * Run all database migrations so that we reach the requested version.
+ * {@link https://github.com/forwards-long-jump/discotron/wiki/Database-migrations|More info on writing migrations.}
+ * @param {string|null} version Which version to migrate to. If null, the latest version is chosen.
+ * @param {boolean} allowDown If true, allow downgrading database version. Otherwise (default), we throw an error.
+ */
+module.exports.doDatabaseMigrations = async (version = null, allowDown = false) => {
+    // While listDiff already handles this similarly, we need a valid "version" string for the SQL as well
+    if (version === null) {
+        version = migrations.latestMigration();
+    }
+
+    const { exec } = require("../apis/database-crud.js");
+
+    const current = await migrations.current();
+    try {
+        const { names, type } = migrations.listDiff(current, version);
+        for (let name of names) {
+            if (type === "down" && allowDown !== true) {
+                Logger.err("May not downgrade without allowDown set to true.");
+                return;
+            } else {
+                // Open migration file as module and retrieve the required function
+                const migration = require(__dirname + "/../migrations/" + name);
+                const func = migration[type];
+                if (func) {
+                    // Execute in database
+                    await exec(func());
+                }
+            }
+        }
+
+        // Write the new current version to the database
+        await exec(`INSERT OR REPLACE INTO _Migrations(name, value) VALUES('version', '${version}');`);
+    } catch (e) {
+        Logger.err(`Error migrating from database version "${current}" to "${version}".
+Ensure you run migrations to downgrade the database before removing migration files, or you risk a broken database state.`);
```

Comment:
This error will print on every launch with no way to "fix" the issue besides ignoring it (and hoping for the application to handle it), or manually changing the migration version (which will probably break your database after more migrations).

Should we add some way reminding the app that the user is aware of this issue and not show it anymore?

I would say no, because it is dangerous to desync database from what's expected by the application. (That's also why running migrations on e.g. pull is probably a better idea, to avoid forgetting migrations!)

Reply:
Message could probably tell the user to "fix missing migrations", but otherwise I think this is fine (this needs better documentation on the wiki to upgrade / downgrade a discotron instance)